src/modules/select/wallets/hyperpay.ts

```diff
@@ -0,0 +1,39 @@
+import { extensionInstallMessage } from '../content'
+import { WalletModule, Helpers, CommonWalletOptions } from '../../../interfaces'
+
+import hyerpayIcon from '../wallet-icons/icon-hyperpay.png'
+import hyerpayIcon2x from '../wallet-icons/icon-hyperpay@2x.png'
+
+function hyperpay(options: CommonWalletOptions): WalletModule {
+  const { preferred, label, iconSrc, svg } = options
+
+  return {
+    name: label || 'HyperPay',
+    iconSrc: iconSrc || hyerpayIcon,
+    iconSrcSet: iconSrc || hyerpayIcon2x,
+    svg,
+    wallet: async (helpers: Helpers) => {
+      const { getProviderName, createModernProviderInterface } = helpers
+
+      const provider =
+        (window as any).ethereum ||
+        ((window as any).web3 && (window as any).web3.currentProvider)
+
+      return {
+        provider,
+        interface:
+          provider && getProviderName(provider) === undefined
```

Comment:
```suggestion
          provider && getProviderName(provider) === 'HyperPay'
```

This check needs to confirm that the injected provider is the HyperPay provider, so it needs to match with the string defined in `src/utilities.ts` on line 223.